lib/pbench/server/templates.py

```diff
@@ -158,103 +249,90 @@ def __init__(
         self,
         prefix: str = None,
         mappings: Path = None,
-        settings: Path = None,
-        base: "TemplateFile" = None,
+        settings: JsonFile = None,
+        skeleton: JsonFile = None,
         tool: str = None,
-        skeleton: bool = False,
     ):
         """
         Describe an Elasticsearch template including a JSON document mappings
-        file, a JSON settings file, or both.
+        file and a JSON settings file, which will be merged into a full
+        Elasticsearch template payload.
 
-        A "base" template may also be linked, from which a final template will
-        be built using shared settings and optionally base mappings.
+        A "skeleton" JSON file may also be linked, from which the mappings will
+        be built by merging tool-specific properties and version into a common
+        base template.
 
         Args:
             prefix: Pbench index prefix string
             mappings: JSON document description. Defaults to None.
             settings: JSON index settings. Defaults to None.
-            base: Referenced base settings and/or mappings. Defaults to None.
             tool: Tool mapping files stitch together mapping JSON from the
-                main and base template objects using a parsed "tool" name.
-            skeleton: The JSON mappings file does not have a version.
+                main and skeleton templates using a parsed "tool" name.
+            skeleton: A versionless skeleton mappings file to merge with the tool
+                mappings.
         """
         self.prefix = prefix
-        self.mappings = JsonFile(mappings) if mappings else None
-        self.settings = JsonFile(settings) if settings else None
-        if mappings:
-            if tool:
-                self.key = tool
-                m = self._fpat.match(mappings.name)
-                self.name = m.group("toolname")
-            else:
-                self.name = mappings.stem if mappings else None
-                self.key = self.name
-            self.index_info = self.index_patterns[self.key] if not skeleton else None
+        self.settings = settings
+        if tool:
+            self.mappings = JsonToolFile(mappings)
+            self.key = tool
+            m = self._fpat.match(mappings.name)
+            self.name = m.group("toolname")
         else:
-            # If this is just a holder for a shared settings file, there's not
-            # much else we can do with it.
-            self.key = None
-            self.name = None
-            self.index_info = None
-        self.base = base
+            self.mappings = JsonFile(mappings)
+            self.name = mappings.stem
+            self.key = self.name
```

Comment:
Does this code behave well if `mappings` is `None`?

Reply:
Probably not; but perhaps this would be more clear if I really do reject attempts to construct a `TemplateFile` with missing parameters. Sigh.